Qapple/Qapple/SourceCode/Data/Service/Network/NetworkClient.swift

```diff
@@ -0,0 +1,189 @@
+}
+
+/// 네트워킹을 수행할 클라이언트 객체
+struct NetworkClient {
+    
+    /// SignIn 요청을 수행합니다.
+    func signIn<T: Decodable, U: Encodable>(url: URL, body: U) async throws -> T {
+        do {
+            
+            // 네트워킹 수행(토큰 X)
+            var request = URLRequest(url: url)
+            request.httpMethod = "POST"
+            request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+            
+            request.httpBody = try JSONEncoder().encode(body)
+            
+            let (data, response) = try await URLSession.shared.data(for: request)
+            
+            return try decodeResponse(data: data, response: response)
+        } catch {
+            throw error
+        }
+    }
+    
+    /// GET 요청을 수행합니다.
+    func get<T>(url: URL) async throws -> T where T: Decodable {
+        do {
+            
+            // 네트워킹 수행
+            let (data, response) = try await request(url: url, method: "GET")
+            
+            return try decodeResponse(data: data, response: response)
+        } catch {
+            throw error
+        }
+    }
+    
+    /// POST 요청을 수행합니다.
+    func post<T: Decodable, U: Encodable>(url: URL, body: U) async throws -> T {
+        do {
+            
+            // 네트워킹 수행
+            let (data, response) = try await request(url: url, body: body, method: "POST")
+            
+            return try decodeResponse(data: data, response: response)
+        } catch {
+            throw error
+        }
+    }
+    
+    /// FETCH 요청을 수행합니다.
+    func fetch<T: Decodable, U: Encodable>(url: URL, body: U) async throws -> T {
+        do {
+            
+            // 네트워킹 수행
+            let (data, response) = try await request(url: url, body: body, method: "FETCH")
+            
+            return try decodeResponse(data: data, response: response)
+        } catch {
+            throw error
+        }
+    }
+    
+    /// DELETE 요청을 수행합니다.
+    func delete<T>(url: URL) async throws -> T where T: Decodable {
+        do {
+            
+            // 네트워킹 수행
+            let (data, response) = try await request(url: url, method: "DELETE")
+            
+            return try decodeResponse(data: data, response: response)
+        } catch {
+            throw error
+        }
+    }
+    
+}
+
+// MARK: - URLSession
+
+extension NetworkClient {
+    
+    /// URLSession을 통한 네트워킹을 수행합니다.
+    ///
+    /// - 커스텀 에러(NetworkError)를 반환하기 위해, 재사용을 염두에 두고 함수로 감싸주었습니다.
+    func request(url: URL, body: Encodable? = nil, method: String) async throws -> (Data, URLResponse) {
+        do {
+            var request = URLRequest(url: url)
+            request.httpMethod = method
+            request.setValue( // TODO: 실제 토큰으로 변경
+                "Bearer eyJhbGciOiJIUzUxMiJ9.eyJ0b2tlblR5cGUiOiJhY2Nlc3MiLCJtZW1iZXJJZCI6MTU0LCJyb2xlIjoiQUNBREVNSUVSIiwiaWF0IjoxNzM2OTE1ODMyLCJleHAiOjE3Mzc3Nzk4MzJ9.x2qe-8kZ3wyl0JGsNOfCmKoioEXoZS9H5liwG2A1bWDgeJk3JPkPPKm86Hbu5B_VD1UK5nJ_JxwQm1wTen04Hg",
```

Comment:
++ 이대로면 토큰 값이 그대로 develop에 머지될 것 같아 요렇게 바꿔주시면 좋을 것 같아요!

```suggestion
            request.setValue(
                try SignInInfo.shared.token(.access),
```

Reply:
확인했습니다!